### PR TITLE
Don't attempt to regenerate keys if they appear to have already been generated.

### PR DESCRIPTION
**Pull request by MattSGS**

### CODE REVIEW

**Comment by peterschwarz** [comment]
You need to do a couple of things with this commit: 

* Add a sign-off.  This is as easy as running `git commit --amend -s`
* Follow the [community contributing guide](https://sawtooth.hyperledger.org/docs/core/releases/latest/community/contributing.html) around commit messages (specifically, we link to this post by [Chris Beam](https://chris.beams.io/posts/git-commit/#seven-rules)).  You can also make this modifications with the above `git commit --amend -s`
* You'll need to go through the DCO steps if you haven't already, which are under the Checks.

File: docker/compose/sawtooth-default.yaml [hunk 1]
```diff
@@ -72,6 +76,6 @@ services:
     depends_on:
       - rest-api
     entrypoint: "bash -c \"\
-        sawtooth keygen && \
+        (test -e /etc/sawtooth/keys/validator.priv || sawtooth keygen) && \
```
Comment (rbuysse) [inline docker/compose/sawtooth-default.yaml]: I think this should be testing for the user key at `/root/.sawtooth/keys/root.priv`

Comment (aludvik) [reply docker/compose/sawtooth-default.yaml]: @rbuysse is correct. `sawtooth keygen` generates `$HOME/.sawtooth/keys/$USER.priv` and `sawadm keygen` generates `/etc/sawtooth/keys/validator.priv`

**Comment by dcmiddle** [comment]
Thanks again for the fix @MattSGS .  Looks like there were a few fixups required including the sign-off. If we don't hear back soon I'll assume you've moved on to help other projects and I will close this for you.

**Comment by vaporos** [comment]
Closing due to lack of DCO sign-off. If fixed, please re-open.

**Pull request closed without merging**